### PR TITLE
Switch to exifr for image metadata extraction

### PR DESCRIPTION
**Pull request by paescuj**
Replace outdated [exif-reader](https://github.com/devongovett/exif-reader) and [icc](https://github.com/lovell/icc) libraries as well as the util `parse-iptc.ts` by the [exifr](https://github.com/MikeKovarik/exifr) library which provides all required functionalities (even more) to extract metadata from images.

This will fix the following errors which have appeared recently (not sure what caused it, though - see e.g. https://github.com/directus/directus/actions/runs/947627163):
```
Could not find a declaration file for module 'exif-reader'. '/home/runner/work/directus/directus/node_modules/exif-reader/index.js' implicitly has an 'any' type.
```
```
Could not find a declaration file for module 'icc'. '/home/runner/work/directus/directus/node_modules/icc/index.js' implicitly has an 'any' type.
```

---

Note:
I also wanted to replace `sharp` in  `api/src/services/files.ts` (only used to get image dimensions here) by something more lightweight like [image-size](https://github.com/image-size/image-size) or [probe-image-size](https://github.com/nodeca/probe-image-size), but those libraries doesn't seem to be ready for us (non-async / no types declaration).

### CODE REVIEW

File: api/src/services/files.ts [hunk 1]
```diff
@@ -78,37 +76,22 @@ export class FilesService extends ItemsService {
 				payload.height = meta.height;
 			}
 
-			payload.filesize = meta.size;
```
Comment (paescuj) [inline api/src/services/files.ts:81]: payload.filesize is already set on line 65, no need to overwrite it again

**Comment by paescuj** [comment]
Ups, totally forgot about migrations... Not ready to be merged.

**Comment by rijkvanzanten** [comment]
Also, for what it's worth, those two issues you mentioned in the OP is a little bug in the shims that I already fixed under aggregate/insights 🙂 

https://github.com/directus/directus/blob/main/api/src/types/shims.d.ts

Apparently imports need to be in the declare block instead of top level

CleanShot 2021-06-18 at 10 05 52@2x

**Comment by paescuj** [comment]
> Also, for what it's worth, those two issues you mentioned in the OP is a little bug in the shims that I already fixed under aggregate/insights 🙂
> 
> https://github.com/directus/directus/blob/main/api/src/types/shims.d.ts
> 
> Apparently imports need to be in the declare block instead of top level
> 
> CleanShot 2021-06-18 at 10 05 52@2x

Ah, nice 👍 I thought that it must be something like that, but I decided to go the hard way 😄

Besides of this, how do you feel about this change? Should I look into migrations or should I rather drop this pull request?

**Comment by paescuj** [comment]
Btw., it seems like you're planing to merge the aggregate/insights branch soon? 😄🥳

**Comment by rijkvanzanten** [comment]
Youuuuu know it!

**Comment by rijkvanzanten** [comment]
What's the difference in the exif format between the two libraries?

**Comment by paescuj** [comment]
exif-reader outputs "idf0" data under "image" while exifr outputs it under "idf0":
```js
// exif-reader
{
  image: {
    Make: "Google",
    Model: "Pixel",
   [...]

// exifr
{
  ifd0: {
    Make: "Google",
    Model: "Pixel",
   [...]

**Comment by paescuj** [comment]
Closing in favor of #6922

**Pull request closed without merging**